skxray/core/correlation.py

```diff
@@ -253,6 +253,211 @@ def multi_tau_auto_corr(num_levels, num_bufs, labels, images):
     return g2, lag_steps
 
 
+def multi_tau_auto_corr_partial_data(num_levels, num_bufs, labels, images, previous):
+    """
+    This function computes one-time correlations.
+
+    It uses a scheme to achieve long-time correlations inexpensively
+    by downsampling the data, iteratively combining successive frames.
+
+    The longest lag time computed is num_levels * num_bufs.
+
+    Parameters
+    ----------
+    num_levels : int
+        how many generations of downsampling to perform, i.e.,
+        the depth of the binomial tree of averaged frames
+
+    num_bufs : int, must be even
+        maximum lag step to compute in each generation of
+        downsampling
+
+    labels : array
+        labeled array of the same shape as the image stack;
+        each ROI is represented by a distinct label (i.e., integer)
+
+    images : iterable of 2D arrays
+        dimensions are: (rr, cc)
+
+    Returns
+    -------
+    g2 : array
+        matrix of normalized intensity-intensity autocorrelation
+        shape (num_levels, number of labels(ROI))
+
+    lag_steps : array
+        delay or lag steps for the multiple tau analysis
+        shape num_levels
+
+    Notes
+    -----
+
+    The normalized intensity-intensity time-autocorrelation function
+    is defined as
+
+    :math ::
+        g_2(q, t') = \frac{<I(q, t)I(q, t + t')> }{<I(q, t)>^2}
+
+    ; t' > 0
+
+    Here, I(q, t) refers to the scattering strength at the momentum
+    transfer vector q in reciprocal space at time t, and the brackets
+    <...> refer to averages over time t. The quantity t' denotes the
+    delay time
+
+    This implementation is based on code in the language Yorick
+    by Mark Sutton, based on published work. [1]_
+
+    References
+    ----------
+
+    .. [1] D. Lumma, L. B. Lurio, S. G. J. Mochrie and M. Sutton,
+        "Area detector based photon correlation in the regime of
+        short data batches: Data reduction for dynamic x-ray
+        scattering," Rev. Sci. Instrum., vol 70, p 3274-3289, 2000.
+
+    """
+    # In order to calculate correlations for `num_bufs`, images must be
+    # kept for up to the maximum lag step. These are stored in the array
+    # buffer. This algorithm only keeps number of buffers and delays but
+    # several levels of delays number of levels are kept in buf. Each
+    # level has twice the delay times of the next lower one. To save
+    # needless copying, of cyclic storage of images in buf is used.
+
+    if previous is None:
+        if num_bufs % 2 != 0:
```

Comment:
Yes same block is need for two time correlation, So I have taken this out and made small helper function for that. I think this can use that too